spyder/plugins/updatemanager/workers.py

```diff
@@ -95,59 +87,161 @@ class AssetInfo(TypedDict):
+
+    Returns
+    -------
+    checksum : str | None
+        Checksum for the provided asset. None is returned if the checksum is
+        not available for the provided asset name.
+    """
+    logger.info(f"Getting checksum from {url}")
+    page = requests.get(url, headers=GH_HEADERS)
+    page.raise_for_status()
+
+    digest = page.text.strip().split("\n")
+    data = {k: v for v, k in [item.split() for item in digest]}
+    checksum = data.get(name, None)
+
+    logger.info(f"Checksum for {name}: {checksum}")
+
+    return checksum
+
+
+def get_asset_info(release_info: dict) -> None | AssetInfo:
+    """
+    Get the version, name, update type, download URL, and size for the asset
+    of the given release.
+
+    Parameters
+    ----------
+    release_info: dict
+        Release information from Github for a single release
+
+    Returns
+    -------
+    asset_info: None | AssetInfo
         Information about the asset.
     """
-    if isinstance(release, str):
-        release = parse(release)
+    release = parse(release_info["tag_name"])
 
-    if CURRENT_VERSION.major < release.major:
+    if CURRENT_VERSION.major < release.major or not is_conda_based_app():
         update_type = UpdateType.Major
     elif CURRENT_VERSION.minor < release.minor:
         update_type = UpdateType.Minor
     else:
         update_type = UpdateType.Micro
 
-    mach = platform.machine().lower().replace("amd64", "x86_64")
-
     if update_type == UpdateType.Major or not is_conda_based_app():
+        ext = platform.machine().lower().replace("amd64", "x86_64")
         if os.name == 'nt':
-            plat, ext = 'Windows', 'exe'
+            ext += '.exe'
         if sys.platform == 'darwin':
-            plat, ext = 'macOS', 'pkg'
+            ext += '.pkg'
         if sys.platform.startswith('linux'):
-            plat, ext = 'Linux', 'sh'
-        name = f'Spyder-{plat}-{mach}.{ext}'
+            ext += '.sh'
     else:
-        name = 'spyder-conda-lock.zip'
+        ext = '.zip'
+
+    asset_info = AssetInfo(version=release, update_type=update_type)
+    for asset in release_info["assets"]:
+        if asset["name"].endswith(ext):
+            asset_info.update(
+                filename=asset["name"],
+                url=asset["browser_download_url"]
+            )
+        if asset["name"] == "Spyder-checksums.txt":
+            asset_info.update(
+                checksum_url=asset["browser_download_url"]
+            )
+        if asset_info.get("url") and asset_info.get("checksum_url"):
+            break
+
+    if asset_info.get("url") is None or asset_info.get("checksum_url") is None:
+        # Both assets are required
+        asset_info = None
 
-    url = (
-        'https://github.com/spyder-ide/spyder/releases/download/'
-        f'v{release}/{name}'
-    )
+    return asset_info
 
-    return AssetInfo(filename=name, update_type=update_type, url=url)
+
+def validate_download(file, checksum):
```

Comment:
```suggestion
def validate_download(file: str, checksum: str) -> bool:
    """
    Validate sha256 checksum of the provided file against provided the checksum.

    Parameters
    ----------
    file : str
        Full path to the file to be verified.
    checksum : str
        sha256 checksum to match against the computed checksum.

    Returns
    -------
    valid : bool
        True if the file's computed checksum matches the provided checksum,
        False otherwise.
    """
```